tests/mmm/test_transformers.py

```diff
@@ -148,6 +149,20 @@ def test_geometric_adstock_good_alpha(self, x, alpha, l_max):
         assert y_np[1] == x[1] + alpha * x[0]
         assert y_np[2] == x[2] + alpha * x[1] + (alpha**2) * x[0]
 
+    @pytest.mark.parametrize(
+        "x, alpha, l_max",
+        [
+            (np.ones(shape=(100)), -0.3, 10),
+            (np.ones(shape=(100)), -2, 10),
+            (np.ones(shape=(100)), 22.5, 10),
+            (np.ones(shape=(100)), 2, 10),
+        ],
+    )
+    def test_geometric_adstock_bad_alpha(self, x, alpha, l_max):
+        y = geometric_adstock(x=x, alpha=alpha, l_max=l_max)
```

Comment:
As x is the same for all cases I think it is better to put `x=np.ones(shape=(100)` inside the `test_geometric_adstock_bad_alpha` and just have 

```python
    @pytest.mark.parametrize(
        "alpha, l_max",
    . . . 
```